pkg/api/helpers.go

```diff
@@ -79,19 +79,42 @@ var Semantic = conversion.EqualitiesOrDie(
 	},
 )
 
-var standardResources = sets.NewString(
-	string(ResourceMemory),
+var standardComputeResources = sets.NewString(
```

Comment:
splitting standard and compute resources seems arbitrary. Why not, instead, make a set for each check, e.g. integerResources, nonnegativeResources, etc.

Memory should also probably be required to be an integer number of bytes, but this may break configs. (1.4GiB is actually not an integer number of bytes!)